### PR TITLE
add bounce keyboard

### PR DESCRIPTION
**Pull request by moyi4681**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kbdfans/bounce/pad/pad.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_numpad( \
```
Comment (drashna) [inline keyboards/kbdfans/bounce/pad/pad.h:20]:
```suggestion
#define LAYOUT_numpad_6x4( \
```

File: keyboards/kbdfans/bounce/pad/info.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+    "keyboard_name": "bouncepad", 
+    "maintainer": "moyi4681", 
+    "layouts": {
+        "LAYOUT_numpad": {
```
Comment (drashna) [inline keyboards/kbdfans/bounce/pad/info.json:5]:
```suggestion
        "LAYOUT_numpad_6x4": {
```

File: keyboards/kbdfans/bounce/pad/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_numpad( /* Base */
```
Comment (drashna) [inline keyboards/kbdfans/bounce/pad/keymaps/default/keymap.c:20]:
```suggestion
  [0] = LAYOUT_numpad_6x4( /* Base */
```

File: keyboards/kbdfans/bounce/pad/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    KC_ESC,  KC_LCTL, KC_LALT, KC_BSPC,
+    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+    KC_P7,   KC_P8,   KC_P9,
+    KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
+    KC_P1,   KC_P2,   KC_P3, 
+             KC_P0,   KC_PDOT, KC_PENT
+  ),
+  [1] = LAYOUT_numpad(
```
Comment (drashna) [inline keyboards/kbdfans/bounce/pad/keymaps/default/keymap.c:28]:
```suggestion
  [1] = LAYOUT_numpad_6x4(
```

File: keyboards/kbdfans/bounce/pad/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    _______, _______, _______, _______,
+    _______, _______, _______, _______,
+    _______, _______, _______, 
+    _______, _______, _______, _______,
+    _______, _______, _______, 
+             _______, _______, _______
+  ),
+  [2] = LAYOUT_numpad(
```
Comment (drashna) [inline keyboards/kbdfans/bounce/pad/keymaps/default/keymap.c:36]:
```suggestion
  [2] = LAYOUT_numpad_6x4(
```

File: keyboards/kbdfans/bounce/pad/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    _______, _______, _______, _______,
+    _______, _______, _______, _______,
+    _______, _______, _______, 
+    _______, _______, _______, _______,
+    _______, _______, _______, 
+             _______, _______, _______
+  ),
+  [3] = LAYOUT_numpad(
```
Comment (drashna) [inline keyboards/kbdfans/bounce/pad/keymaps/default/keymap.c:44]:
```suggestion
  [3] = LAYOUT_numpad_6x4(
```

File: keyboards/kbdfans/bounce/pad/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_numpad( /* Base */
```
Comment (drashna) [inline keyboards/kbdfans/bounce/pad/keymaps/via/keymap.c:20]:
```suggestion
  [0] = LAYOUT_numpad_6x4( /* Base */
```

File: keyboards/kbdfans/bounce/pad/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    KC_ESC,  KC_LCTL, KC_LALT, KC_BSPC,
+    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+    KC_P7,   KC_P8,   KC_P9,
+    KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
+    KC_P1,   KC_P2,   KC_P3, 
+             KC_P0,   KC_PDOT, KC_PENT
+  ),
+  [1] = LAYOUT_numpad(
```
Comment (drashna) [inline keyboards/kbdfans/bounce/pad/keymaps/via/keymap.c:28]:
```suggestion
  [1] = LAYOUT_numpad_6x4(
```

File: keyboards/kbdfans/bounce/pad/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    _______, _______, _______, _______,
+    _______, _______, _______, _______,
+    _______, _______, _______, 
+    _______, _______, _______, _______,
+    _______, _______, _______, 
+             _______, _______, _______
+  ),
+  [2] = LAYOUT_numpad(
```
Comment (drashna) [inline keyboards/kbdfans/bounce/pad/keymaps/via/keymap.c:36]:
```suggestion
  [2] = LAYOUT_numpad_6x4(
```

File: keyboards/kbdfans/bounce/pad/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    _______, _______, _______, _______,
+    _______, _______, _______, _______,
+    _______, _______, _______, 
+    _______, _______, _______, _______,
+    _______, _______, _______, 
+             _______, _______, _______
+  ),
+  [3] = LAYOUT_numpad(
```
Comment (drashna) [inline keyboards/kbdfans/bounce/pad/keymaps/via/keymap.c:44]:
```suggestion
  [3] = LAYOUT_numpad_6x4(
```

File: keyboards/kbdfans/bounce/pad/rules.mk [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no       # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
```
Comment (drashna) [inline keyboards/kbdfans/bounce/pad/rules.mk:18]:
```suggestion
AUDIO_ENABLE = no           # Audio output

LAYOUTS = numpad_6x4
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/bounce/75/hotswap/readme.md [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+
+A customizable 75% HOTSWAP keyboard.
+
+* Keyboard Maintainer: [moyi4681](https://github.com/moyi4681)
+* Hardware Supported: KBDFANS
+* Hardware Availability: [kbdfans](https://kbdfans.myshopify.com/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make kbdfans/bonce/75/hotswap:default
```
Comment (fauxpark) [inline keyboards/kbdfans/bounce/75/hotswap/readme.md:11]:
```suggestion
    make kbdfans/bounce/75/hotswap:default
```

File: keyboards/kbdfans/bounce/75/soldered/readme.md [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+
+A customizable 75% HOTSWAP keyboard.
+
+* Keyboard Maintainer: [moyi4681](https://github.com/moyi4681)
+* Hardware Supported: KBDFANS
+* Hardware Availability: [kbdfans](https://kbdfans.myshopify.com/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make kbdfans/bonce/75/soldered:default
```
Comment (fauxpark) [inline keyboards/kbdfans/bounce/75/soldered/readme.md:11]:
```suggestion
    make kbdfans/bounce/75/soldered:default
```

File: keyboards/kbdfans/bounce/pad/readme.md [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+
+A customizable pad keyboard, include hotswap and soldered version.
+
+* Keyboard Maintainer: [moyi4681](https://github.com/moyi4681)
+* Hardware Supported: KBDFANS
+* Hardware Availability: [kbdfans](https://kbdfans.myshopify.com/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make kbdfans/bonce/pad:default
```
Comment (fauxpark) [inline keyboards/kbdfans/bounce/pad/readme.md:11]:
```suggestion
    make kbdfans/bounce/pad:default
```

File: keyboards/kbdfans/bounce/75/hotswap/config.h [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4B42
+#define PRODUCT_ID      0x7001
+#define DEVICE_VER      0x0003
+#define MANUFACTURER    KBDFANS
+#define PRODUCT         BONCE75_HOTSWAP
```
Comment (fauxpark) [inline keyboards/kbdfans/bounce/75/hotswap/config.h:26]:
```suggestion
#define PRODUCT         BOUNCE75_HOTSWAP
```

File: keyboards/kbdfans/bounce/75/soldered/config.h [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4B42
+#define PRODUCT_ID      0x7000
+#define DEVICE_VER      0x0003
+#define MANUFACTURER    KBDFANS
+#define PRODUCT         BONCE75_SOLDERED
```
Comment (fauxpark) [inline keyboards/kbdfans/bounce/75/soldered/config.h:26]:
```suggestion
#define PRODUCT         BOUNCE75_SOLDERED
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/bounce/pad/keymaps/default/keymap.c (hunk 6 above)
Comment (drashna) [reply keyboards/kbdfans/bounce/pad/keymaps/default/keymap.c:44]: This still needs applying.

**Review by drashna — APPROVED** [review]

File: keyboards/kbdfans/bounce/75/soldered/soldered.h [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include "quantum.h"
+#define XXX KC_NO
```
Comment (zvecr) [inline keyboards/kbdfans/bounce/75/soldered/soldered.h:19]:
```suggestion
```

File: keyboards/kbdfans/bounce/75/hotswap/hotswap.h [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include "quantum.h"
+#define XXX KC_NO
```
Comment (zvecr) [inline keyboards/kbdfans/bounce/75/hotswap/hotswap.h:19]:
```suggestion
```

File: keyboards/kbdfans/bounce/pad/keymaps/default/keymap.c [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+             _______, _______, _______
+  ),
+  [3] = LAYOUT_numpad_6x4(
+    _______, _______, _______, _______,
+    _______, _______, _______, _______,
+    _______, _______, _______, 
+    _______, _______, _______, _______,
+    _______, _______, _______, 
+             _______, _______, _______
+  ),
```
Comment (zvecr) [inline keyboards/kbdfans/bounce/pad/keymaps/default/keymap.c:51]:
```suggestion
```

**Review by zvecr — APPROVED** [review]

**Pull request merged**